pkg/controller/datavolume-controller.go

```diff
@@ -575,6 +575,14 @@ func (r *DatavolumeReconciler) getSnapshotClassForSmartClone(dataVolume *cdiv1.D
 		return "", errors.New("source PVC not found")
 	}
 
+	sourceVolumeMode := GetVolumeMode(pvc.Spec.VolumeMode)
+	targetVolumeMode := GetVolumeMode(dataVolume.Spec.PVC.VolumeMode)
+	if sourceVolumeMode != targetVolumeMode {
+		r.log.V(3).Info("Source PVC and target PVC has different volume modes, falling back to host assisted clone", "source volume mode",
+			sourceVolumeMode, "target volume mode", targetVolumeMode)
+		return "", errors.New("Source PVC and target PVC has different volume modes, falling back to host assisted clone")
```

Comment:
I think It should be "Source PVC and target PVC ha[ve]" here and two lines above. Maybe some native English speaker can confirm